clang/test/SemaSYCL/sycl-device-intel-max-global-work-dim-template.cpp

```diff
@@ -58,17 +58,80 @@ int main() {
 
 // Test that checks template parameter support on function.
 template <int N>
-// expected-error@+1{{'max_global_work_dim' attribute requires a non-negative integral compile time constant expression}}
+// expected-error@+1{{'max_global_work_dim' attribute requires integer constant between 0 and 3 inclusive}}
 [[intel::max_global_work_dim(N)]] void func3() {}
 
+// Test that checks template instantiations for different argument values.
+template <int size>
+[[intel::max_global_work_dim(1)]] void func4(); // expected-note {{previous attribute is here}}
+
+template <int size>
+[[intel::max_global_work_dim(size)]] void func4() {} // expected-warning {{attribute 'max_global_work_dim' is already applied with different arguments}}
+
+// Checks correctness of mutual usage of different work_group_size attributes:
+// reqd_work_group_size, max_work_group_size, and max_global_work_dim.
+// In case the value of 'max_global_work_dim' attribute equals to 0 we shall
+// ensure that if max_work_group_size and reqd_work_group_size attributes exist,
+// they hold equal values (1, 1, 1).
+template <int N>
+[[intel::max_work_group_size(N, N, N)]] void func5(); // expected-error {{'max_work_group_size' X-, Y- and Z- sizes must be 1 when 'max_global_work_dim' attribute is used with value 0}}
```

Comment:
We don't have to touch the diagnostic now, but those dashes after the attribute parameter names look odd to me and I think they should be removed. WDYT?

Reply:
> We don't have to touch the diagnostic now, but those dashes after the attribute parameter names look odd to me and I think they should be removed. WDYT?

Thanks @AaronBallman for catching this. Yes, i agree with you that this seems very odd. I will update this in this PR since we are removing all old codes related to this diagnostic here.